app/views/result_filters/provider/new.html.erb

```diff
@@ -10,7 +10,10 @@
     <ul class="govuk-list govuk-list--bullet">
       <% @provider_suggestions.each do |provider| %>
         <li data-qa="provider_suggestion">
-          <%= govuk_link_to "#{provider.provider_name} (#{provider.provider_code})", results_path(filter_params.merge(query: provider.provider_name)), { class: "govuk-link", data: { qa: "provider_suggestion__link" } } %>
+          <%= govuk_link_to "#{provider.provider_name} (#{provider.provider_code})", 
+            results_path(remove_previous_parameters(filter_params.merge(query: provider.provider_name))), 
```

Comment:
Might be worth moving this to a helper method - it's quite hard to read as it is here

Reply:
I do agree with @danielburnley 

`remove_previous_parameters` & `filter_params` are both from FilterParameters`

Reply:
How about `filter_params_without_previous` to combine `remove_previous_parameters` and `filter_params`?